iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/HMSTablePropertyHelper.java

```diff
@@ -218,15 +155,14 @@ public static SortOrder getSortOrder(Properties props, Schema schema) {
    */
   public static Properties getCatalogProperties(org.apache.hadoop.hive.metastore.api.Table hmsTable) {
     Properties properties = new Properties();
-
-    hmsTable.getParameters().entrySet().stream().filter(e -> e.getKey() != null && e.getValue() != null).forEach(e -> {
+    Map<String, String> hmsTableParameters = Optional.ofNullable(hmsTable.getParameters()).orElseGet(Maps::newHashMap);
```

Comment:
can it be null?

Reply:
Yes, it can be null. Added safety around it in several places in `HiveIcebergMetaHook`, `HiveIcebergStorageHandler` and `IcebergTableUtil`.